markets/bfp-market/test/integration/modules/PerpAccountModule.mergeAccounts.test.ts

```diff
@@ -637,6 +637,65 @@ describe('PerpAccountModule mergeAccounts', () => {
     );
   });
 
+  it('should update debt correction when merging', async () => {
+    const { BfpMarketProxy, MergeAccountSettlementHookMock } = systems();
+    const { fromTrader } = await createAccountsToMerge();
+
+    const { collateral, market, collateralDepositAmount, marketId } = await depositMargin(
+      bs,
+      genTrader(bs, {
+        desiredTrader: fromTrader,
+      })
+    );
+    const marketDigest1 = await BfpMarketProxy.getMarketDigest(marketId);
+
+    assertBn.isZero(marketDigest1.debtCorrection);
+
+    const order = await genOrder(bs, market, collateral, collateralDepositAmount, {
+      desiredHooks: [MergeAccountSettlementHookMock.address],
+    });
+    const { receipt } = await commitAndSettle(bs, marketId, fromTrader, order);
+    const orderSettleEvent = findEventSafe(receipt, 'OrderSettled', BfpMarketProxy);
+
+    const marketDigest = await BfpMarketProxy.getMarketDigest(marketId);
+
+    // Calculate expected debtCorrection.
+    // `debtCorrection` = `prevDebtCorrection` + `fundingDelta` + `notionalDelta` + `totalPositionPnl`
+    // First calculate `debtCorrection` for the first order
+    const prevDebtCorrection = wei(0);
+    const sizeDelta = orderSettleEvent.args.sizeDelta;
+    const fundingDelta = wei(0);
+    const notionalDelta = wei(orderSettleEvent.args.fillPrice).mul(sizeDelta).toBN();
+    const totalPositionPnl = wei(0);
+
+    const expectedDebtCorrectionBeforeMerge = prevDebtCorrection
+      .add(fundingDelta)
+      .add(notionalDelta)
+      .add(totalPositionPnl);
+
+    // Calculate `debtCorrection` from realizing `from` position.
+    const sizeDelta2 = wei(0).sub(orderSettleEvent.args.sizeDelta);
```

Comment:
Minor: Non-blocking but consider a `calcDebtCorrection` helper function and add it to `calculations.ts`